packages/amplify_datastore/example/integration_test/relationship_type_test.dart

```diff
@@ -0,0 +1,366 @@
+        hasManyEvent = Amplify.DataStore.observe(HasManyModel.classType).first;
+      });
+
+      testWidgets('precondition', (WidgetTester tester) async {
+        var queriedChildren =
+            await Amplify.DataStore.query(HasManyChildModel.classType);
+        expect(queriedChildren, isEmpty);
+        var queriedParents =
+            await Amplify.DataStore.query(HasManyModel.classType);
+        expect(queriedParents, isEmpty);
+      });
+
+      testWidgets('save parent', (WidgetTester tester) async {
+        await Amplify.DataStore.save(parent);
+        var parents = await Amplify.DataStore.query(HasManyModel.classType);
+        expect(parents, isNotEmpty);
+      });
+
+      testWidgets('save children', (WidgetTester tester) async {
+        for (var child in children) {
+          await Amplify.DataStore.save(child);
+        }
+        var queriedChildren =
+            await Amplify.DataStore.query(HasManyChildModel.classType);
+        expect(queriedChildren, isNotEmpty);
+      });
+
+      testWidgets('query parent', (WidgetTester tester) async {
+        var parents = await Amplify.DataStore.query(HasManyModel.classType);
+        var queriedParent = parents.first;
+        expect(queriedParent, parent);
+        expect(queriedParent.id, parent.id);
+        expect(queriedParent.name, parent.name);
+      });
+
+      testWidgets('query children', (WidgetTester tester) async {
+        var queriedChildren =
+            await Amplify.DataStore.query(HasManyChildModel.classType);
+        for (var i = 0; i < 5; i++) {
+          var queriedChild = queriedChildren[i];
+          var actualChild = children[i];
+          expect(queriedChild, actualChild);
+          expect(queriedChild.parent, actualChild.parent);
+        }
+      });
+
+      testWidgets('observe parent', (WidgetTester tester) async {
+        var event = await hasManyEvent;
+        var observedParent = event.item;
+        expect(observedParent.id, parent.id);
+        expect(observedParent.name, parent.name);
+      });
+
+      testWidgets('observe children', (WidgetTester tester) async {
+        var event = await childEvent;
+        var observedChild = event.item;
+        var firstChild = children.first;
+        expect(observedChild, firstChild);
+        expect(observedChild.parent, firstChild.parent);
+      });
+
+      testWidgets('delete parent', (WidgetTester tester) async {
+        await Amplify.DataStore.delete(parent);
+        var parents = await Amplify.DataStore.query(HasManyModel.classType);
+        expect(parents, isEmpty);
+      });
+
+      testWidgets('delete children', (WidgetTester tester) async {
+        for (var child in children) {
+          await Amplify.DataStore.delete(child);
+        }
+        var queriedChildren =
+            await Amplify.DataStore.query(HasManyChildModel.classType);
+        expect(queriedChildren, isEmpty);
+      });
+    });
+    group('hasMany (cascade save)', () {
```

Comment:
Nit for consistency:
```suggestion

    group('HasMany (cascade save)', () {
```